haas/ext/switches/_console.py

```diff
@@ -119,6 +119,13 @@ def revert_port(self, port):
         self.exit_if_prompt()
         self.console.expect(self.config_prompt)
 
+    def _should_save(self, switch_type):
+        """checks the config file to see if switch should save or not"""
+
+        switch_ext = 'haas.ext.switches.' + switch_type
+        return (not cfg.has_option(switch_ext, 'save') or
```

Comment:
I'm still not a fan - even though I agree that it is elegant, readable is more important in order to avoid .
It's too bad we haven't transitioned to python 3 yet -- we could just use the [fallback](https://docs.python.org/3/library/configparser.html#fallback-values) function in the new ConfigParser.